crates/kitsune_p2p/dht_arc/src/peer_view/peer_view_beta.rs

```diff
@@ -0,0 +1,292 @@
+use crate::*;
+
+/// The default, and only, strategy for generating a PeerView
+#[derive(Debug, Clone, Copy)]
+pub struct PeerStratBeta {
+    pub focus_nearby: bool,
+    pub min_sample_size: u16,
+    pub coverage_buffer: f64,
+    pub total_coverage_buffer: f64,
+    pub default_uptime: f64,
+    pub noise_threshold: f64,
+    pub delta_scale: f64,
+    pub delta_threshold: f64,
+}
+
+impl Default for PeerStratBeta {
+    fn default() -> Self {
+        Self {
+            focus_nearby: true,
+            min_sample_size: DEFAULT_REDUNDANCY_TARGET as u16,
+            coverage_buffer: DEFAULT_COVERAGE_BUFFER,
+            total_coverage_buffer: DEFAULT_TOTAL_COVERAGE_BUFFER,
+            default_uptime: DEFAULT_UPTIME,
+            noise_threshold: DEFAULT_NOISE_THRESHOLD,
+            delta_scale: DEFAULT_DELTA_SCALE,
+            delta_threshold: DEFAULT_DELTA_THRESHOLD,
+        }
+    }
+}
+
+impl PeerStratBeta {
+    pub fn view(&self, arc: DhtArc, peers: &[DhtArc]) -> PeerViewBeta {
+        let peers: Vec<DhtArc> = peers
+            .iter()
+            .filter(|a| arc.contains(a.center_loc))
+            .copied()
+            .collect();
+        Self::view_unchecked(self, arc, peers.as_slice())
+    }
+
+    pub fn view_unchecked(&self, arc: DhtArc, peers: &[DhtArc]) -> PeerViewBeta {
+        let total_coverage = total_coverage(peers);
+        let count = peers.len();
+        let mut full_view = PeerViewBeta::new(*self, arc, total_coverage, count);
+        if let Some(focused_view) =
+            self.check_focused_view(arc, peers, self.min_sample_size as usize)
+        {
+            full_view.focused_view_target = Some(focused_view);
+        }
+        full_view
+    }
+
+    /// Check a more focused view of the network if we are currently viewing a much larger
+    /// portion of the network then we might need to.
+    ///
+    /// The idea here is that you may already have enough peer information to make a
+    /// better extrapolation of the full network in a area more focused around your center
+    /// point. In this case it is not worth waiting for all the peer information to
+    /// sync for the larger view as that might take a long time.
+    ///
+    /// Imagine the case where you join a large network with a full arc, you might have a
+    /// good sample size close to your center long before you sync all the networks peer
+    /// information.
+    fn check_focused_view(&self, arc: DhtArc, peers: &[DhtArc], focus_size: usize) -> Option<f64> {
```

Comment:
I'm not sure I agree with the philosophy of this function... The only property to me that seems to make for an accurate extrapolation would be completeness of data, and we're making no effort to be more complete in the area closer to us than further away. And if we *do* have complete data for the full area locally, is there any benefit in not using it?